File: net/src/main/java/com/example/net/NetApiService.java
package com.example.net;

import android.database.Observable;

import java.util.HashMap;


import okhttp3.ResponseBody;
import retrofit2.http.FieldMap;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.HeaderMap;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.QueryMap;
import retrofit2.http.Url;

public interface NetApiService {
    @GET("{path}")
    io.reactivex.Observable<ResponseBody> getData(@HeaderMap HashMap<String, String> headers, @Path("path") String path, @QueryMap HashMap<String, String> params);
    @POST("{path}")
    @FormUrlEncoded
    io.reactivex.Observable<ResponseBody> postData(@FieldMap HashMap<String, String> headers, @Path("path") String path, @FieldMap HashMap<String, String> params);

}

File: net/src/main/java/com/bw/net/NetApiService.kt
package com.bw.net

import io.reactivex.Observable
import okhttp3.ResponseBody
import retrofit2.http.*

interface NetApiService {

    //ResponseBody 是一个通用的返回类型，如果不清楚服务端返回的json数据，可以通过它来获取
    @GET("path")
    fun getData(@HeaderMap headerMap: HashMap<String,String>,@Path("path") path:String,@QueryMap params:HashMap<String,String>) : Observable<ResponseBody>

    @GET
    fun getMyDate(@Url url:String) : Observable<ResponseBody>

    @POST("path")
    @FormUrlEncoded
    fun postData(@Url url: String,@FieldMap name:HashMap<String,String>,@FieldMap password:HashMap<String,String>,@FieldMap phone:HashMap<String,String>)

}